base/compiler/typeinfer.jl

```diff
@@ -191,6 +191,19 @@ function store_backedges(frame::InferenceState)
                 end
             end
         end
+        edges = frame.src.edges
+        if edges !== nothing
+            Core.println(edges === nothing)
+            Core.println(edges)
+            Core.println(typeof(edges))
+            edges = edges::Vector{MethodInstance}
```

Comment:
should be a `Vector{Any}`, since edges only sometimes end at a MethodInstance

Reply:
In the Cassette use case, don't they always end at a MethodInstance (because Cassette by nature gets concrete values from the generated function)?